cmd/runtimetest/main.go

```diff
@@ -630,6 +630,54 @@ func validateMountsExist(spec *rspec.Spec) error {
 	return nil
 }
 
+func validateMountsOrder(spec *rspec.Spec) error {
+	if runtime.GOOS == "windows" {
+		return nil
+	}
+
+	mountInfos, err := mount.GetMounts()
+	if err != nil {
+		return err
+	}
+
+	type mountOrder struct {
+		Order  int
+		Root   string
+		Dest   string
+		Source string
+	}
+	mountsMap := make(map[string][]mountOrder)
+	for i, mountInfo := range mountInfos {
+		m := mountOrder{
+			Order:  i,
+			Root:   mountInfo.Root,
+			Dest:   mountInfo.Mountpoint,
+			Source: mountInfo.Source,
+		}
+		mountsMap[mountInfo.Mountpoint] = append(mountsMap[mountInfo.Mountpoint], m)
+	}
+	current := -1
+	for _, specMount := range spec.Mounts {
```

Comment:
As a cheap local part of the in-flight #266 you could use `configMount` instead of `specMount` for this variable.